src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpInteractiveCommands.cs

```diff
@@ -0,0 +1,135 @@
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpInteractiveCommands : AbstractInteractiveWindowTest
+    {
+        public CSharpInteractiveCommands(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory)
+        {
+        }
+
+        [Fact]
+        public void VerifyPreviousAndNextHistory()
+        {
+            SubmitText("1 + 2");
+            SubmitText("1.ToString()");
+            SendKeys(new KeyPress(VirtualKey.Up, ShiftState.Alt));
+            VerifyLastReplInput("1.ToString()");
+            SendKeys(VirtualKey.Enter);
+            WaitForReplOutput("\"1\"");
+            SendKeys(new KeyPress(VirtualKey.Up, ShiftState.Alt));
+            VerifyLastReplInput("1.ToString()");
+            SendKeys(new KeyPress(VirtualKey.Up, ShiftState.Alt));
+            VerifyLastReplInput("1 + 2");
+            SendKeys(VirtualKey.Enter);
+            WaitForReplOutput("3");
+            SendKeys(new KeyPress(VirtualKey.Down, ShiftState.Alt));
+            VerifyLastReplInput("1.ToString()");
+            SendKeys(VirtualKey.Enter);
+            WaitForReplOutput("\"1\"");
+        }
+
+        [Fact]
+        public void VerifyMaybeExecuteInput()
+        {
+            InsertCode("2 + 3");
+            SendKeys(VirtualKey.Enter);
+            WaitForReplOutput("5");
+        }
+
+        [Fact]
+        public void VerifyNewLineAndIndent()
+        {
+            InsertCode("3 + ");
+            SendKeys(VirtualKey.Enter);
+            InsertCode("4");
+            SendKeys(VirtualKey.Enter);
+            WaitForReplOutput("7");
+        }
+
+        [Fact]
+        public void VerifyExecuteInput()
+        {
+            SubmitText("1 + ");
+            VerifyLastReplOutputContains("CS1733");
+        }
+
+        [Fact]
+        public void VerifyForceNewLineAndIndent()
+        {
+            InsertCode("1 + 2");
+            SendKeys(VirtualKey.Enter);
+            SubmitText("+ 3");
+            VerifyReplPromptConsistency("<![CDATA[1 + 2 + 3]]>", "6");
+        }
+
+        [Fact]
+        public void VerifyCancelInput()
+        {
+            InsertCode("1 + 4");
+            SendKeys(new KeyPress(VirtualKey.Enter, ShiftState.Shift));
+            SendKeys(VirtualKey.Escape);
+            VerifyLastReplInput(string.Empty);
+        }
+
+        [Fact]
+        public void VerifyUndoAndRedo()
+        {
+            ClearReplText();
+            InsertCode(" 2 + 4 ");
+            SendKeys(new KeyPress(VirtualKey.Z, ShiftState.Ctrl));
+            VerifyReplPromptConsistency("< ![CDATA[]] >", string.Empty);
+            SendKeys(new KeyPress(VirtualKey.Y, ShiftState.Ctrl));
+            VerifyLastReplInput(" 2 + 4 ");
+            SendKeys(VirtualKey.Enter);
+            WaitForReplOutput("6");
+        }
+
+        [Fact]
+        public void CutDeletePasteSelectAll()
+        {
+            SendKeys("Text");
+            ExecuteCommand("Edit.LineStart");
```

Comment:
nit: It would be nice if we had these commands centralized somewhere for general use.